### DISCUSSION TITLE
Prisma db pull TLS connection error

### DISCUSSION THREAD

**Discussion by dhakoby** [prisma/prisma#14223] [Q&A] [answered]
### Error Description

I am using docker base image node:16. When I am building image running npx prisma db pull, I am getting this error.

> [ 9/12] RUN npx prisma db pull:
#13 2.181 Prisma schema loaded from prisma/schema.prisma
#13 2.183 Environment variables loaded from .env
#13 2.215 Datasource "db" - SQL Server
#13 2.245
#13 2.248 - Introspecting based on datasource defined in prisma/schema.prisma
#13 3.256 ✖ Introspecting based on datasource defined in prisma/schema.prisma
#13 3.256
#13 3.257 Error: P1011
#13 3.257
#13 3.257 Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: unexpected EOF)
#13 3.257
------
executor failed running [/bin/sh -c npx prisma db pull]: exit code: 1

But works well with image node:16-alpine

### How to reproduce

Use nestjs application with prisma in a docker base image node:16. And try to build image.

### Expected behavior

It should run and connect to database successfully .

### Prisma schema file
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}
```
### Dockerfile
```dockerfile
FROM node:16

ARG PORT=8080

ENV PORT=8080

WORKDIR /usr/src/app

COPY package*.json ./


RUN npm install


# Copy local directory into container filesystem
COPY . /usr/src/app



#prisma
RUN npx prisma db pull
RUN npx prisma generate

#build typescript to javascript
RUN npm run build



# Make it available.
EXPOSE ${PORT}/tcp

CMD ["node","./dist/main.js"]
```
### Environment & setup

- OS: Windows 11
- Docker v20.10.17
- Node v16

### Prisma version

"prisma": "^4.0.0",
"@prisma/client": "^4.0.0",

**Comment by austincrim** [comment, marked as answer]
Hey @DDavidD1 👋,

Could you share your connection string (sensitive info redacted)?

**Reply by dhakoby** [reply]
I'd like to, but it's a private info. For the info, encrypt is true and trustCerificate also is true)

**Reply by austincrim** [reply]
Do you have `trustCertificate=true` or `trustServerCertificate=true` ?

**Reply by dhakoby** [reply]
Sorry, my bad. Here are the exact parameters I've set to true. encrypt=true; trustServerCertificate=true;
But I figured out what's the problem is. I've changed openssl config file under /etc/ssl/openssl.cnf. Prepended this code: 
openssl_conf = default_conf
[ default_conf ]
ssl_conf = ssl_sect
[ssl_sect]
system_default = system_default_sect
[system_default_sect]
MinProtocol = TLSv1.2
CipherString = DEFAULT:@SECLEVEL=1

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
